pop-api/src/v0/assets/pallets/assets.rs

```diff
@@ -0,0 +1,492 @@
+	#[codec(index = 28)]
+	SetMinBalance { id: AssetIdParameter, min_balance: BalanceParameter },
+	#[codec(index = 29)]
+	TouchOther { id: AssetIdParameter, who: MultiAddress<AccountId, ()> },
+	#[codec(index = 30)]
+	RefundOther { id: AssetIdParameter, who: MultiAddress<AccountId, ()> },
+	#[codec(index = 31)]
+	Block { id: AssetIdParameter, who: MultiAddress<AccountId, ()> },
+}
+
+// TODO: Not being used atm but necessary if we want to provide access to the
+//  rest of the pallet, outside of the use cases.
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Encode, scale::Decode)]
+#[cfg_attr(feature = "std", derive(scale_info::TypeInfo))]
+pub enum AssetsError {
+	/// Account balance must be greater than or equal to the transfer amount.
+	BalanceLow,
+	/// The account to alter does not exist.
+	NoAccount,
+	/// The signing account has no permission to do the operation.
+	NoPermission,
+	/// The given asset ID is unknown.
+	Unknown,
+	/// The origin account is frozen.
+	Frozen,
+	/// The asset ID is already taken.
+	InUse,
+	/// Invalid witness data given.
+	BadWitness,
+	/// Minimum balance should be non-zero.
+	MinBalanceZero,
+	/// Unable to increment the consumer reference counters on the account. Either no provider
+	/// reference exists to allow a non-zero balance of a non-self-sufficient asset, or one
+	/// fewer then the maximum number of consumers has been reached.
+	UnavailableConsumer,
+	/// Invalid metadata given.
+	BadMetadata,
+	/// No approval exists that would allow the transfer.
+	Unapproved,
+	/// The source account would not survive the transfer and it needs to stay alive.
+	WouldDie,
+	/// The asset-account already exists.
+	AlreadyExists,
+	/// The asset-account doesn't have an associated deposit.
+	NoDeposit,
+	/// The operation would result in funds being burned.
+	WouldBurn,
+	/// The asset is a live asset and is actively being used. Usually emit for operations such
+	/// as `start_destroy` which require the asset to be in a destroying state.
+	LiveAsset,
+	/// The asset is not live, and likely being destroyed.
+	AssetNotLive,
+	/// The asset status is not the expected status.
+	IncorrectStatus,
+	/// The asset should be frozen before the given operation.
+	NotFrozen,
+	/// Callback action resulted in error.
+	CallbackFailed,
+}
+
+impl TryFrom<u32> for AssetsError {
+	type Error = PopApiError;
+
+	fn try_from(status_code: u32) -> core::result::Result<Self, Self::Error> {
+		use AssetsError::*;
+		match status_code {
+			0 => Ok(BalanceLow),
+			1 => Ok(NoAccount),
+			2 => Ok(NoPermission),
+			3 => Ok(Unknown),
+			4 => Ok(Frozen),
+			5 => Ok(InUse),
+			6 => Ok(BadWitness),
+			7 => Ok(MinBalanceZero),
+			8 => Ok(UnavailableConsumer),
+			9 => Ok(BadMetadata),
+			10 => Ok(Unapproved),
+			11 => Ok(WouldDie),
+			12 => Ok(AlreadyExists),
+			13 => Ok(NoDeposit),
+			14 => Ok(WouldBurn),
+			15 => Ok(LiveAsset),
+			16 => Ok(AssetNotLive),
+			17 => Ok(IncorrectStatus),
+			18 => Ok(NotFrozen),
+			_ => todo!(),
```

Comment:
This should return an error and not panic.

Reply:
Not required anymore